src/Analyzer/LambdaNode.cpp

```diff
@@ -46,6 +48,11 @@ void LambdaNode::dumpTreeImpl(WriteBuffer & buffer, FormatState & format_state,
     getExpression()->dumpTreeImpl(buffer, format_state, indent + 4);
 }
 
+DataTypePtr LambdaNode::getResultType() const
+{
+    return result_type ? result_type : std::make_shared<DataTypeNothing>();
```

Comment:
It's better to correctly handle `untuple` case in `QueryAnalyzer::resolveFunction`.

Reply:
You're right. I thought there would be more special cases, but it's only untuple and grouping AFAICS, and grouping works fine, so I'll handle it for `untuple` directly instead.